pkg/oc/cli/admin/migrate/migrator.go

```diff
@@ -724,33 +724,50 @@ func DefaultRetriable(info *resource.Info, err error) error {
 // listed and updated. Note that this may return some virtual resources (like imagestreamtags) that can be otherwise
 // represented.
 // TODO: add a field to APIResources for "virtual" (or that points to the canonical resource).
-// TODO: fallback to the scheme when discovery is not possible.
-func FindAllCanonicalResources(d discovery.ServerResourcesInterface, m meta.RESTMapper) ([]schema.GroupResource, error) {
+func FindAllCanonicalResourcesForGroup(d discovery.DiscoveryInterface, m meta.RESTMapper, groupSpecification string) ([]schema.GroupResource, error) {
 	set := make(map[schema.GroupResource]struct{})
-	all, err := d.ServerResources()
+
+	// this call doesn't fail on aggregated apiserver failures
+	groups, err := d.ServerGroups()
 	if err != nil {
 		return nil, err
 	}
-	for _, serverResource := range all {
-		gv, err := schema.ParseGroupVersion(serverResource.GroupVersion)
+	versionsToCheck := []string{}
+	for _, group := range groups.Groups {
+		// match on "*" and "*.<groupSuffix>"
+		if groupSpecification == "*" || strings.HasSuffix(group.Name, groupSpecification[2:]) {
+			for _, version := range group.Versions {
+				versionsToCheck = append(versionsToCheck, version.GroupVersion)
+			}
+		}
+	}
+
+	for _, versionString := range versionsToCheck {
+		gv, err := schema.ParseGroupVersion(versionString)
 		if err != nil {
 			continue
 		}
-		for _, r := range serverResource.APIResources {
+		versionResources, err := d.ServerResourcesForGroupVersion(versionString)
+		if err != nil {
+			return nil, err
```

Comment:
Does this also work correctly during aggregated API failures?

Reply:
The call `d.ServerGroups()` several lines above handles that case, so you're only iterating through available groups. So you should be fine.

Reply:
> Does this also work correctly during aggregated API failures?

Correctly in this case would be "I requested an aggregated group that you confirmed the existence of" and this line says "I must resolve the resources for every version of this group and cannot.  User intent fails, fail command"